plugins/generic/mysqlIndex/SearchDAO.inc.php

```diff
@@ -260,6 +266,40 @@ function insertObject($recordId, $fieldId, &$pos, $date = null, $flush = true) {
 		return $objectId;
 	}
 
+	/**
+	 * Add many record objects to the index (when existent, indexed keywords are cleared).
+	 * Each search object can contain a single date entry; this is an optimization to
+	 * prevent having a separate indexed table to contain dates. If a date is specified
+	 * here, the object will be guaranteed to have that date regardless of whether it was
+	 * created or already existed.
+	 * @param $recordId int
+	 * @param $fieldIds array
+	 * @param $date string optional
+	 * @param $flush boolean Whether or not to flush values of an existing object
+	 */
+	function insertObjects($recordId, $fieldIds, $date = null, $flush = true) {
+		if (!count($fieldIds)) {
+			return;
+		}
+
+		$this->update('
+			INSERT IGNORE INTO search_objects (record_id, raw_field_id, object_time)
+			VALUES ' . implode(', ', array_map(function ($fieldId) use ($recordId, $date) {
```

Comment:
The `docs/README` requirements still specify PHP 4.2.x; mind bumping it to something less prehistoric while you're at it (e.g. PHP 5.6)? This syntax will require that.

Reply:
I'm using PHP 7.1 locally and the guys are already using 7.2 in the PKP servers (which is hosting some instances of the Harvester + PKP Index), so maybe we can follow the OJS?

Reply:
That's fine; I don't think there's much benefit to maintaining PHP 5.x support.